fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2784,6 +2798,14 @@ private void addIndexStateReadConflict(@Nonnull String indexName) {
         if (!getRecordMetaData().hasIndex(indexName)) {
             throw new MetaDataException("Index " + indexName + " does not exist in meta-data.");
         }
+        if (indexStateReadConflicts == null) {
+            indexStateReadConflicts = new HashSet<>(8);
```

Comment:
Should this be using a `ConcurrentHashMap`? I think multiple threads can access it at once. Because it's safe to set this multiple times for the same index, having this throw away the full hash map in certain race conditions seems fine, but I'm still worried about messing up internal state in the hash map if there are multiple accessors.

Reply:
Yeah, I debated.  I will modify.